rasa/nlu/components.py

```diff
@@ -460,347 +466,75 @@ def name(self) -> Text:
+        self,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> None:
+        """Initializes a new NLU Component."""
+        self.component_config: Dict[Text, Any] = config
 
         self.partial_processing_pipeline = None
         self.partial_processing_context = None
+        self._model_storage = model_storage
+        self._resource = resource
+        self._execution_context = execution_context
 
-    @classmethod
-    def required_packages(cls) -> List[Text]:
-        """Specifies which python packages need to be installed.
-
-        E.g. ``["spacy"]``. More specifically, these should be
-        importable python package names e.g. `sklearn` and not package
-        names in the dependencies sense e.g. `scikit-learn`
-
-        This list of requirements allows us to fail early during training
-        if a required package is not installed.
-
-        Returns:
-            The list of required package names.
-        """
-        return []
-
-    @classmethod
-    def load(
-        cls,
-        meta: Dict[Text, Any],
-        model_dir: Text,
-        model_metadata: Optional["Metadata"] = None,
-        cached_component: Optional["Component"] = None,
-        **kwargs: Any,
-    ) -> "Component":
-        """Loads this component from file.
-
-        After a component has been trained, it will be persisted by
-        calling `persist`. When the pipeline gets loaded again,
-        this component needs to be able to restore itself.
-        Components can rely on any context attributes that are
-        created by :meth:`components.Component.create`
-        calls to components previous to this one.
-
-        Args:
-            meta: Any configuration parameter related to the model.
-            model_dir: The directory to load the component from.
-            model_metadata: The model's :class:`rasa.nlu.model.Metadata`.
-            cached_component: The cached component.
-
-        Returns:
-            the loaded component
-        """
-        if cached_component:
-            return cached_component
-
-        return cls(meta)
-
-    @classmethod
-    def create(
-        cls, component_config: Dict[Text, Any], config: RasaNLUModelConfig
-    ) -> "Component":
-        """Creates this component (e.g. before a training is started).
-
-        Method can access all configuration parameters.
-
-        Args:
-            component_config: The components configuration parameters.
-            config: The model configuration parameters.
-
-        Returns:
-            The created component.
-        """
-        # Check language supporting
-        language = config.language
-        if not cls.can_handle_language(language):
-            # check failed
-            raise UnsupportedLanguageError(cls.name, language)
-
-        return cls(component_config)
-
-    def provide_context(self) -> Optional[Dict[Text, Any]]:
```

Comment:
Context is needed by components such as the spacy components, so that they can communicate with each other. This will be replaced by distinct edges in the graph.